[1.18] Update CNI to v0.8.6 and build hyperkube:v1.0.0 image

justaugustus opened the pull request:
**What type of PR is this?**

/kind cleanup
/area dependency release-eng
/sig network release

**What this PR does / why we need it**:

Update CNI to v0.8.6

cc: @kubernetes/release-engineering @kubernetes/sig-network-misc 
/assign @BenTheElder @yujuhong @tallclair
ref: https://github.com/kubernetes/release/pull/1309

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update CNI to v0.8.6
```

justaugustus committed:
Update CNI to v0.8.6

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

saschagrunert reviewed (approved):
/lgtm

cpanato reviewed (approved):
/lgtm

cblecker commented:
/hold
Requesting details on why this is being picked back

cblecker commented:
/hold cancel
/approve

liggitt commented:
/approve
/priority important-soon

liggitt commented on build/dependencies.yaml:
```diff
@@ -1,9 +1,7 @@
-    version: 0.8.5
+    version: 0.8.6
     refPaths:
-    - path: build/debian-hyperkube-base/Makefile
-      match: CNI_VERSION=
```
why was this reference removed if the file was not also removed in this PR?

liggitt commented:
/approve cancel

liggitt commented:
question on https://github.com/kubernetes/kubernetes/pull/91387#discussion_r430564695

justaugustus committed:
build: Add dependency entries for debian-hyperkube-base

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus committed:
hyperkube: Build v1.0.0 image

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
> why was this reference removed if the file was not also removed in this PR?

@liggitt -- Good catch. I was attempting to prevent us from having to bump the `debian-hyperkube-base` image, since hyperkube was marked as deprecated.

However, given our discussion, we need to continue maintaining that image until 1.18 is out of support.
I've updated `debian-hyperkube-base` to support K8s Infra image building, use `debian-base:v2.1.0`, and CNI plugins @ v0.8.6 in this PR.

The accompanying k/test-infra PR is here: https://github.com/kubernetes/test-infra/pull/17720
A test image building GCB job is running at https://console.cloud.google.com/cloud-build/builds/59e055d5-b803-46b1-a8ea-24f7856128c9?project=k8s-staging-build-image.

@BenTheElder, @cblecker, @dims, @listx, and I have access to view the build logs.

justaugustus commented:
/hold until https://github.com/kubernetes/test-infra/pull/17720 merges

justaugustus commented:
/hold cancel
(https://github.com/kubernetes/test-infra/pull/17720 has merged)

justaugustus commented:
The test image build was successful:

```shell
$ skopeo inspect docker://gcr.io/k8s-staging-build-image/debian-hyperkube-base@sha256:ebc2e27bd33c4e337b59bc70582dbd3b8593a0933c48219394ae6e20fa1f7243 --raw
```

```json
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:cbb7ef1d3b757a21269e4282ca8a5c88047fec366645f8743bdfc90974b7f043",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:d9f9271baa0806801f826ee4a6c432d22e94628147b78de489e2526aa2859964",
         "platform": {
            "architecture": "arm",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:62d4cc457482a1f2e9f942448b89ff33a73ba4f121f2f230218c5a58da1aa475",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:c69a41bc8e423301fed31aed8899dd124bf974181245832fcb1cc747f55a1dd5",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:e128b94620722aacc6784a257309d75da322a210f45a74557c1df833cf4d6539",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

Here's the old debian-hyperkube-base image:

```shell
$ skopeo inspect docker://k8s.gcr.io/debian-hyperkube-base:0.12.1 --raw
```

```json
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:5d4ea2fb5fbe9a9a9da74f67cf2faefc881968bc39f2ac5d62d9167e575812a1",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1785,
         "digest": "sha256:73260814af61522ff6aa48291df457d3bb0a91c4bf72e7cfa51fbaf03eb65fae",
         "platform": {
            "architecture": "arm",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:78eeb1a31eef7c16f954444d64636d939d89307e752964ad6d9d06966c722da3",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:92857d647abe8d9c7b4d7160cd5699112afc12fde369082a8ed00688b17928a9",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:c11d74fa0538c67238576c247bfaddf95ebaa90cd03cb4d2f2ac3c6ebe0441e2",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

And a comparison between the old and new images:

```
$ time container-diff diff k8s.gcr.io/debian-hyperkube-base@sha256:5d4ea2fb5fbe9a9a9da74f67cf2faefc881968bc39f2ac5d62d9167e575812a1 gcr.io/k8s-staging-build-image/debian-hyperkube-base@sha256:cbb7ef1d3b757a21269e4282ca8a5c88047fec366645f8743bdfc90974b7f043 --type apt
```

<details>

```
-----Apt-----

Packages found only in k8s.gcr.io/debian-hyperkube-base@sha256:5d4ea2fb5fbe9a9a9da74f67cf2faefc881968bc39f2ac5d62d9167e575812a1:
NAME                                   VERSION                       SIZE
-e2fslibs                              1.43.4-2                      449K
-gcc-6-base                            6.3.0-18 deb9u1               209K
-iproute2                              4.9.0-1 deb9u1                1.7M
-iputils-ping                          3:20161105-1                  111K
-libbabeltrace-ctf1                    1.5.1-1                       421K
-libboost-iostreams1.62.0              1.62.0 dfsg-4                 141K
-libboost-program-options1.62.0        1.62.0 dfsg-4                 554K
-libboost-random1.62.0                 1.62.0 dfsg-4                 68K
-libboost-regex1.62.0                  1.62.0 dfsg-4                 1.1M
-libboost-system1.62.0                 1.62.0 dfsg-4                 60K
-libboost-thread1.62.0                 1.62.0 dfsg-4                 205K
-libcephfs1                            10.2.11-2                     7.3M
-libcomerr2                            1.43.4-2                      83K
-libdevmapper-event1.02.1              2:1.02.137-2                  72K
-libevent-2.0-5                        2.0.21-stable-3               341K
-libfcgi0ldbl                          2.4.0-8.4 b1                  458K
-libgdbm3                              1.8.3-14                      68K
-libicu57                              57.1-6 deb9u2                 29.3M
-libidn11                              1.33-1                        306K
-libipset3                             6.30-2                        289K
-liblvm2app2.2                         2.02.168-2                    1.4M
-libncurses5                           6.0 20161126-1+deb9u2         287K
-libncursesw5                          6.0 20161126-1+deb9u2         347K
-libonig4                              6.1.3-2                       701K
-libperl5.24                           5.24.1-3 deb9u5               20.4M
-libprocps6                            2:3.3.12-3 deb9u1             121K
-librgw2                               10.2.11-2                     12.1M
-libssl1.0.2                           1.0.2q-1~deb9u1               3.5M
-libtinfo5                             6.0 20161126-1+deb9u2         478K
-libtirpc1                             0.2.5-1.2 deb9u1              219K
-libunistring0                         0.9.6 really0.9.3-0.1         1.1M
-liburcu4                              0.9.3-1                       257K
-libustr-1.0-1                         1.0.4-6                       258K
-multiarch-support                     2.24-11 deb9u3                220K
-perl-modules-5.24                     5.24.1-3 deb9u5               17.2M
-procps                                2:3.3.12-3 deb9u1             690K
-samba-common                          2:4.5.12 dfsg-2+deb9u4        278K

Packages found only in gcr.io/k8s-staging-build-image/debian-hyperkube-base@sha256:cbb7ef1d3b757a21269e4282ca8a5c88047fec366645f8743bdfc90974b7f043:
NAME                                   VERSION                       SIZE
-bsdutils                              1:2.33.1-0.1                  293K
-fdisk                                 2.33.1-0.1                    483K
-gcc-8-base                            8.3.0-6                       248K
-libboost-atomic1.67.0                 1.67.0-13 deb10u1             2M
-libboost-iostreams1.67.0              1.67.0-13 deb10u1             2.1M
-libboost-program-options1.67.0        1.67.0-13 deb10u1             2.5M
-libboost-regex1.67.0                  1.67.0-13 deb10u1             3M
-libboost-system1.67.0                 1.67.0-13 deb10u1             2M
-libboost-thread1.67.0                 1.67.0-13 deb10u1             2.1M
-libcephfs2                            12.2.11 dfsg1-2.1+b1          1.2M
-libcom-err2                           1.44.5-1 deb10u3              92K
-libevent-2.1-6                        2.1.8-stable-4                427K
-libext2fs2                            1.44.5-1 deb10u3              476K
-libgdbm-compat4                       1.18.1-4                      66K
-libgdbm6                              1.18.1-4                      117K
-libgfapi0                             5.5-3                         2.6M
-libgfchangelog0                       5.5-3                         2.4M
-libgfdb0                              5.5-3                         2.4M
-libgfrpc0                             5.5-3                         2.5M
-libgfxdr0                             5.5-3                         2.5M
-libglusterfs-dev                      5.5-3                         3.3M
-libglusterfs0                         5.5-3                         3.4M
-libgoogle-perftools4                  2.7-1                         1.1M
-libicu63                              63.1-6 deb10u1                30.9M
-libipset11                            6.38-1.2                      207K
-libmpdec2                             2.4.2-2                       254K
-libncurses6                           6.1 20181013-2+deb10u2        334K
-libncursesw6                          6.1 20181013-2+deb10u2        411K
-libnftnl11                            1.1.2-2                       212K
-libonig5                              6.9.1-1                       616K
-libpcre2-8-0                          10.32-5                       576K
-libperl5.28                           5.28.1-6                      25.6M
-libpython2-stdlib                     2.7.16-1                      38K
-libpython3-stdlib                     3.7.3-1                       37K
-libpython3.7                          3.7.3-2 deb10u1               4.9M
-libpython3.7-minimal                  3.7.3-2 deb10u1               3.8M
-libpython3.7-stdlib                   3.7.3-2 deb10u1               8M
-libseccomp2                           2.3.3-4                       306K
-libsnappy1v5                          1.1.7-1                       47K
-libtcmalloc-minimal4                  2.7-1                         497K
-libtinfo6                             6.1 20181013-2+deb10u2        521K
-libtirpc-common                       1.1.4-0.4                     45K
-libtirpc3                             1.1.4-0.4                     254K
-libunistring2                         0.9.10-1                      1.6M
-libunwind8                            1.2.1-9                       188K
-liburcu6                              0.10.2-1                      293K
-libzstd1                              1.3.8 dfsg-3                  670K
-perl-modules-5.28                     5.28.1-6                      18.5M
-python-certifi                        2018.8.24-1                   301K
-python-idna                           2.6-1                         271K
-python-prettytable                    0.7.2-4                       83K
-python2                               2.7.16-1                      136K
-python2-minimal                       2.7.16-1                      144K
-python3                               3.7.3-1                       187K
-python3-certifi                       2018.8.24-1                   300K
-python3-chardet                       3.0.4-3                       411K
-python3-idna                          2.6-1                         271K
-python3-jwt                           1.7.0-2                       86K
-python3-minimal                       3.7.3-1                       121K
-python3-pkg-resources                 40.8.0-1                      534K
-python3-prettytable                   0.7.2-4                       83K
-python3-requests                      2.21.0-1                      223K
-python3-six                           1.12.0-1                      60K
-python3-urllib3                       1.24.1-1                      389K
-python3.7                             3.7.3-2 deb10u1               408K
-python3.7-minimal                     3.7.3-2 deb10u1               9.4M

Version differences:
PACKAGE                         IMAGE1 (k8s.gcr.io/debian-hyperkube-base@sha256:5d4ea2fb5fbe9a9a9da74f67cf2faefc881968bc39f2ac5d62d9167e575812a1)        IMAGE2 (gcr.io/k8s-staging-build-image/debian-hyperkube-base@sha256:cbb7ef1d3b757a21269e4282ca8a5c88047fec366645f8743bdfc90974b7f043)
-adduser                        3.115, 849K                                                                                                              3.118, 849K
-apt                            1.4.9, 3.5M                                                                                                              1.8.2, 4M
-attr                           1:2.4.47-2 b2, 156K                                                                                                      1:2.4.48-4, 160K
-base-files                     9.9 deb9u7, 333K                                                                                                         10.3 deb10u3, 340K
-base-passwd                    3.5.43, 229K                                                                                                             3.5.46, 232K
-bash                           4.4-5, 5.7M                                                                                                              5.0-4, 6.3M
-ca-certificates                20161130 nmu1+deb9u1, 434K                                                                                               20190110, 384K
-ceph-common                    10.2.11-2, 32.1M                                                                                                         12.2.11 dfsg1-2.1+b1, 34M
-cifs-utils                     2:6.7-1, 230K                                                                                                            2:6.8-2, 231K
-conntrack                      1:1.4.4 snapshot20161117-5, 102K                                                                                         1:1.4.5-2, 103K
-coreutils                      8.26-3, 14.7M                                                                                                            8.30-3, 15.4M
-dash                           0.5.8-2.4, 204K                                                                                                          0.5.10.2-5, 212K
-debconf                        1.5.61, 558K                                                                                                             1.5.71, 520K
-debian-archive-keyring         2017.5, 118K                                                                                                             2019.1, 198K
-debianutils                    4.8.1.1, 213K                                                                                                            4.8.6.1, 226K
-diffutils                      1:3.5-3, 1.3M                                                                                                            1:3.7-3, 1.5M
-dmsetup                        2:1.02.137-2, 236K                                                                                                       2:1.02.155-3, 253K
-dpkg                           1.18.25, 6.6M                                                                                                            1.19.7, 6.5M
-e2fsprogs                      1.43.4-2, 3.9M                                                                                                           1.44.5-1 deb10u3, 1.4M
-ebtables                       2.0.10.4-3.5 b1, 351K                                                                                                    2.0.10.4 snapshot20181205-3, 265K
-ethtool                        1:4.8-1 b1, 372K                                                                                                         1:4.19-1, 393K
-findutils                      4.6.0 git+20161106-2, 1.8M                                                                                               4.6.0 git+20190209-2, 1.9M
-fuse                           2.9.7-1 deb9u2, 136K                                                                                                     2.9.9-1 deb10u1, 140K
-git                            1:2.11.0-3 deb9u4, 28.2M                                                                                                 1:2.20.1-2 deb10u3, 34.6M
-git-man                        1:2.11.0-3 deb9u4, 1.4M                                                                                                  1:2.20.1-2 deb10u3, 1.6M
-glusterfs-client               3.8.8-1, 3.6M                                                                                                            5.5-3, 2.4M
-glusterfs-common               3.8.8-1, 16.9M                                                                                                           5.5-3, 15.8M
-gpgv                           2.1.18-8~deb9u3, 721K                                                                                                    2.2.12-1 deb10u1, 837K
-grep                           2.27-2, 1.1M                                                                                                             3.3-1, 1014K
-gzip                           1.6-5 b1, 231K                                                                                                           1.9-3, 243K
-hostname                       3.18 b1, 47K                                                                                                             3.21, 54K
-init-system-helpers            1.48, 131K                                                                                                               1.56 nmu1, 133K
-ipset                          6.30-2, 140K                                                                                                             6.38-1.2, 150K
-iptables                       1.6.0 snapshot20161117-6, 1.5M                                                                                           1.8.2-4, 2.5M
-jq                             1.5 dfsg-1.3, 100K                                                                                                       1.5 dfsg-2+b1, 101K
-keyutils                       1.5.9-9, 126K                                                                                                            1.6-6, 146K
-kmod                           23-2, 220K                                                                                                               26-1, 237K
-libacl1                        2.2.52-3 b1, 62K                                                                                                         2.2.53-4, 70K
-libacl1-dev                    2.2.52-3 b1, 222K                                                                                                        2.2.53-4, 230K
-libaio1                        0.3.110-3, 30K                                                                                                           0.3.112-3, 35K
-libapt-pkg5.0                  1.4.9, 3M                                                                                                                1.8.2, 3.2M
-libattr1                       1:2.4.47-2 b2, 42K                                                                                                       1:2.4.48-4, 55K
-libattr1-dev                   1:2.4.47-2 b2, 101K                                                                                                      1:2.4.48-4, 86K
-libaudit-common                1:2.6.7-2, 30K                                                                                                           1:2.8.4-3, 33K
-libaudit1                      1:2.6.7-2, 150K                                                                                                          1:2.8.4-3, 161K
-libbabeltrace1                 1.5.1-1, 132K                                                                                                            1.5.6-2 deb10u1, 548K
-libblkid1                      2.29.2-1 deb9u1, 367K                                                                                                    2.33.1-0.1, 428K
-libbsd0                        0.8.3-1, 163K                                                                                                            0.9.1-2, 191K
-libbz2-1.0                     1.0.6-8.1, 96K                                                                                                           1.0.6-9.2~deb10u1, 104K
-libc-bin                       2.24-11 deb9u3, 3.3M                                                                                                     2.28-10, 3.4M
-libc-dev-bin                   2.24-11 deb9u3, 377K                                                                                                     2.28-10, 405K
-libc6                          2.24-11 deb9u3, 10.4M                                                                                                    2.28-10, 12M
-libc6-dev                      2.24-11 deb9u3, 15.3M                                                                                                    2.28-10, 18.5M
-libcap-ng0                     0.7.7-3 b1, 43K                                                                                                          0.7.9-2, 47K
-libcap2                        1:2.25-1, 47K                                                                                                            1:2.25-2, 52K
-libcurl3-gnutls                7.52.1-5 deb9u8, 616K                                                                                                    7.64.0-4 deb10u1, 689K
-libdb5.3                       5.3.28-12 deb9u1, 1.8M                                                                                                   5.3.28 dfsg1-0.5, 1.8M
-libdebconfclient0              0.227, 67K                                                                                                               0.249, 72K
-libdevmapper1.02.1             2:1.02.137-2, 401K                                                                                                       2:1.02.155-3, 473K
-libdw1                         0.168-1, 751K                                                                                                            0.176-1.1, 893K
-libedit2                       3.1-20160903-3, 241K                                                                                                     3.1-20181209-1, 250K
-libelf1                        0.168-1, 934K                                                                                                            0.176-1.1, 932K
-liberror-perl                  0.17024-1, 65K                                                                                                           0.17027-2, 76K
-libexpat1                      2.2.0-2 deb9u1, 369K                                                                                                     2.2.6-2 deb10u1, 519K
-libfdisk1                      2.29.2-1 deb9u1, 469K                                                                                                    2.33.1-0.1, 546K
-libffi6                        3.2.1-6, 56K                                                                                                             3.2.1-9, 56K
-libfuse2                       2.9.7-1 deb9u2, 341K                                                                                                     2.9.9-1 deb10u1, 350K
-libgcc1                        1:6.3.0-18 deb9u1, 108K                                                                                                  1:8.3.0-6, 116K
-libgcrypt20                    1.7.6-2 deb9u3, 1.2M                                                                                                     1.8.4-5, 1.3M
-libglib2.0-0                   2.50.3-2, 4.9M                                                                                                           2.58.3-2 deb10u2, 3.6M
-libgmp10                       2:6.1.2 dfsg-1, 568K                                                                                                     2:6.1.2 dfsg-4, 565K
-libgnutls30                    3.5.8-5 deb9u4, 2.4M                                                                                                     3.6.7-4 deb10u3, 2.6M
-libgpg-error0                  1.26-2, 572K                                                                                                             1.35-1, 175K
-libgssapi-krb5-2               1.15-1 deb9u1, 423K                                                                                                      1.17-3, 428K
-libhogweed4                    3.3-1 b2, 232K                                                                                                           3.4.1-1, 245K
-libibverbs1                    1.2.1-2, 104K                                                                                                            22.1-1, 158K
-libidn2-0                      0.16-1 deb9u1, 174K                                                                                                      2.0.5-1 deb10u1, 280K
-libip4tc0                      1.6.0 snapshot20161117-6, 110K                                                                                           1.8.2-4, 120K
-libip6tc0                      1.6.0 snapshot20161117-6, 114K                                                                                           1.8.2-4, 120K
-libiptc0                       1.6.0 snapshot20161117-6, 88K                                                                                            1.8.2-4, 98K
-libjq1                         1.5 dfsg-1.3, 329K                                                                                                       1.5 dfsg-2+b1, 330K
-libk5crypto3                   1.15-1 deb9u1, 311K                                                                                                      1.17-3, 313K
-libkeyutils1                   1.5.9-9, 36K                                                                                                             1.6-6, 45K
-libkmod2                       23-2, 116K                                                                                                               26-1, 129K
-libkrb5-3                      1.15-1 deb9u1, 1M                                                                                                        1.17-3, 1.1M
-libkrb5support0                1.15-1 deb9u1, 159K                                                                                                      1.17-3, 170K
-libldap-2.4-2                  2.4.44 dfsg-5+deb9u2, 509K                                                                                               2.4.47 dfsg-3+deb10u2, 525K
-libldap-common                 2.4.44 dfsg-5+deb9u2, 108K                                                                                               2.4.47 dfsg-3+deb10u2, 112K
-liblz4-1                       0.0~r131-2 b1, 93K                                                                                                       1.8.3-1, 141K
-liblzma5                       5.2.2-1.2 b1, 339K                                                                                                       5.2.4-1, 263K
-libmount1                      2.29.2-1 deb9u1, 403K                                                                                                    2.33.1-0.1, 475K
-libnetfilter-conntrack3        1.0.6-2, 134K                                                                                                            1.0.7-1, 138K
-libnettle6                     3.3-1 b2, 358K                                                                                                           3.4.1-1, 380K
-libnfnetlink0                  1.0.1-3, 61K                                                                                                             1.0.1-3 b1, 41K
-libnghttp2-14                  1.18.1-1, 201K                                                                                                           1.36.0-2 deb10u1, 210K
-libnl-3-200                    3.2.27-2, 176K                                                                                                           3.4.0-1, 177K
-libnl-route-3-200              3.2.27-2, 462K                                                                                                           3.4.0-1, 543K
-libnspr4                       2:4.12-6, 331K                                                                                                           2:4.20-1, 319K
-libnss3                        2:3.26.2-1.1 deb9u1, 3.7M                                                                                                2:3.42.1-1 deb10u2, 3.6M
-libp11-kit0                    0.23.3-2, 438K                                                                                                           0.23.15-2, 1.3M
-libpam-modules                 1.1.8-3.6, 874K                                                                                                          1.3.1-5, 1M
-libpam-modules-bin             1.1.8-3.6, 220K                                                                                                          1.3.1-5, 238K
-libpam-runtime                 1.1.8-3.6, 1016K                                                                                                         1.3.1-5, 1M
-libpam0g                       1.1.8-3.6, 229K                                                                                                          1.3.1-5, 244K
-libpcre3                       2:8.39-3, 668K                                                                                                           2:8.39-12, 673K
-libpopt0                       1.16-10 b2, 224K                                                                                                         1.16-12, 222K
-libpsl5                        0.17.0-3, 73K                                                                                                            0.20.2-2, 90K
-libpython-stdlib               2.7.13-2, 37K                                                                                                            2.7.16-1, 38K
-libpython2.7                   2.7.13-2 deb9u3, 3.5M                                                                                                    2.7.16-2 deb10u1, 3.3M
-libpython2.7-minimal           2.7.13-2 deb9u3, 2.7M                                                                                                    2.7.16-2 deb10u1, 2.7M
-libpython2.7-stdlib            2.7.13-2 deb9u3, 8.4M                                                                                                    2.7.16-2 deb10u1, 8.7M
-librados2                      10.2.11-2, 6.5M                                                                                                          12.2.11 dfsg1-2.1+b1, 11.8M
-libradosstriper1               10.2.11-2, 9.1M                                                                                                          12.2.11 dfsg1-2.1+b1, 1.2M
-librbd1                        10.2.11-2, 8.4M                                                                                                          12.2.11 dfsg1-2.1+b1, 3.3M
-librdmacm1                     1.1.0-2, 134K                                                                                                            22.1-1, 197K
-libreadline5                   5.2 dfsg-3+b1, 339K                                                                                                      5.2 dfsg-3+b13, 343K
-libreadline7                   7.0-3, 416K                                                                                                              7.0-5, 416K
-librtmp1                       2.4 20151223.gitfa8646d.1-1+b1, 142K                                                                                     2.4 20151223.gitfa8646d.1-2, 141K
-libsasl2-2                     2.1.27~101-g0780600 dfsg-3, 183K                                                                                         2.1.27 dfsg-1+deb10u1, 188K
-libsasl2-modules-db            2.1.27~101-g0780600 dfsg-3, 96K                                                                                          2.1.27 dfsg-1+deb10u1, 101K
-libselinux1                    2.6-3 b3, 209K                                                                                                           2.8-1 b1, 194K
-libsemanage-common             2.6-2, 39K                                                                                                               2.8-2, 30K
-libsemanage1                   2.6-2, 291K                                                                                                              2.8-2, 301K
-libsepol1                      2.6-2, 653K                                                                                                              2.8-1, 724K
-libsmartcols1                  2.29.2-1 deb9u1, 257K                                                                                                    2.33.1-0.1, 314K
-libsqlite3-0                   3.16.2-5 deb9u1, 1.1M                                                                                                    3.27.2-3, 1.3M
-libss2                         1.43.4-2, 95K                                                                                                            1.44.5-1 deb10u3, 104K
-libssh2-1                      1.7.0-1, 262K                                                                                                            1.8.0-2.1, 270K
-libssl1.1                      1.1.0j-1~deb9u1, 3.5M                                                                                                    1.1.1d-0 deb10u3, 4M
-libstdc++6                     6.3.0-18 deb9u1, 2M                                                                                                      8.3.0-6, 2M
-libsystemd0                    232-25 deb9u8, 654K                                                                                                      241-7~deb10u3, 767K
-libtalloc2                     2.1.8-1, 102K                                                                                                            2.1.14-2, 107K
-libtasn1-6                     4.10-1.1 deb9u1, 112K                                                                                                    4.13-3, 112K
-libudev1                       232-25 deb9u8, 224K                                                                                                      241-7~deb10u4, 259K
-libuuid1                       2.29.2-1 deb9u1, 107K                                                                                                    2.33.1-0.1, 120K
-libwbclient0                   2:4.5.12 dfsg-2+deb9u4, 190K                                                                                             2:4.9.5 dfsg-5+deb10u1, 214K
-libwrap0                       7.6.q-26, 104K                                                                                                           7.6.q-28, 108K
-libxml2                        2.9.4 dfsg1-2.2+deb9u2, 2.1M                                                                                             2.9.4 dfsg1-7+b3, 1.8M
-libxtables12                   1.6.0 snapshot20161117-6, 135K                                                                                           1.8.2-4, 149K
-linux-libc-dev                 4.9.130-2, 4.4M                                                                                                          4.19.118-2, 5M
-login                          1:4.4-4.1, 2.7M                                                                                                          1:4.5-1.1, 2.6M
-lsb-base                       9.20161125, 49K                                                                                                          10.2019051400, 49K
-mime-support                   3.60, 110K                                                                                                               3.62, 110K
-mount                          2.29.2-1 deb9u1, 444K                                                                                                    2.33.1-0.1, 418K
-netbase                        5.4, 44K                                                                                                                 5.6, 44K
-nfs-common                     1:1.3.4-2.1, 730K                                                                                                        1:1.3.4-2.5, 762K
-openssh-client                 1:7.4p1-10 deb9u4, 4M                                                                                                    1:7.9p1-10 deb10u2, 3.5M
-openssl                        1.1.0j-1~deb9u1, 1.3M                                                                                                    1.1.1d-0 deb10u3, 1.4M
-passwd                         1:4.4-4.1, 2.4M                                                                                                          1:4.5-1.1, 2.5M
-perl                           5.24.1-3 deb9u5, 651K                                                                                                    5.28.1-6, 575K
-perl-base                      5.24.1-3 deb9u5, 7.4M                                                                                                    5.28.1-6, 9.9M
-psmisc                         22.21-2.1 b2, 589K                                                                                                       23.2-1, 637K
-python                         2.7.13-2, 648K                                                                                                           2.7.16-1, 68K
-python-cephfs                  10.2.11-2, 1.4M                                                                                                          12.2.11 dfsg1-2.1+b1, 362K
-python-chardet                 2.3.0-2, 496K                                                                                                            3.0.4-3, 411K
-python-minimal                 2.7.13-2, 145K                                                                                                           2.7.16-1, 43K
-python-pkg-resources           33.1.1-1, 540K                                                                                                           40.8.0-1, 563K
-python-rados                   10.2.11-2, 1.9M                                                                                                          12.2.11 dfsg1-2.1+b1, 895K
-python-rbd                     10.2.11-2, 1.5M                                                                                                          12.2.11 dfsg1-2.1+b1, 552K
-python-requests                2.12.4-1, 511K                                                                                                           2.21.0-1, 227K
-python-six                     1.10.0-3, 56K                                                                                                            1.12.0-1, 60K
-python-urllib3                 1.19.1-1, 302K                                                                                                           1.24.1-1, 389K
-python2.7                      2.7.13-2 deb9u3, 359K                                                                                                    2.7.16-2 deb10u1, 377K
-python2.7-minimal              2.7.13-2 deb9u3, 3.7M                                                                                                    2.7.16-2 deb10u1, 3.6M
-readline-common                7.0-3, 89K                                                                                                               7.0-5, 89K
-rpcbind                        0.2.3-0.6, 144K                                                                                                          1.2.5-0.3 deb10u1, 148K
-sed                            4.4-1, 799K                                                                                                              4.7-1, 883K
-sensible-utils                 0.0.9 deb9u1, 62K                                                                                                        0.0.12, 62K
-socat                          1.7.3.1-2 deb9u1, 1006K                                                                                                  1.7.3.2-2, 1010K
-tar                            1.29b-1.1, 2.7M                                                                                                          1.30 dfsg-6, 2.8M
-ucf                            3.0036, 191K                                                                                                             3.0038 nmu1, 188K
-udev                           232-25 deb9u8, 6.9M                                                                                                      241-7~deb10u4, 8.1M
-util-linux                     2.29.2-1 deb9u1, 3.5M                                                                                                    2.33.1-0.1, 4.2M
-xfsprogs                       4.9.0 nmu1, 4.2M                                                                                                         4.20.0-1, 3.3M
-zlib1g                         1:1.2.8.dfsg-5, 156K                                                                                                     1:1.2.11.dfsg-1, 174K
```

</details>

liggitt commented on build/debian-hyperkube-base/Makefile:
```diff
@@ -17,15 +17,15 @@
+REGISTRY?=gcr.io/k8s-staging-build-image
 IMAGE?=$(REGISTRY)/debian-hyperkube-base
-TAG=0.12.2
+TAG?=v1.0.0
```
is there a reason to make this v1.0.0? will that trick people into using it when they shouldn't? ("It's GA!")

justaugustus replied:
@liggitt -- mostly because there are barely calculable amount of changes between the images and I want to make it clear that there might not be a SemVer-ish compatibility guarantee between package versions in both.

liggitt replied:
ok... I'm fuzzy on the implications of this in a cherry-pick back to 1.17/1.16... would those also update their own hyperkube base images?

justaugustus replied:
This PR is going to do two things:
- update CNI plugins to v0.8.6
- trigger a staging build for debian-hyperkube-base@v1.0.0

Then we'll need to:
- promote the image to prod
- open another 1.18 PR to actually use the new image in the relevant file(s) (specified in `build/dependencies.yaml`)

We should only enable debian-hyperkube-base image building for `release-1.18`, because we don't actually need to build the image on every branch (just the latest branch that still supports hyperkube).

The cherry picks for 1.17 and 1.16 will instead include just:
- update CNI plugins to v0.8.6
- use the new image in the relevant file(s) (specified in `build/dependencies.yaml`)

liggitt replied:
ah, ok. would it make sense to split the hyperkube-base build bit we don't intend to pick back to 1.17/1.16 into a separate PR so this one can include the `actually use the new image in the relevant file(s) (specified in build/dependencies.yaml)` bit and match 1.17/1.16 as closely as possible?

justaugustus replied:
@liggitt -- I actually purposely did it this way.

If we do the image build as a separate PR, it means that we'll have to build that one using CNI plugins @ v0.8.5, which means roughly 5 PRs related to 1.18, instead of 2.

justaugustus replied:
Concretely, the 1.18 PRs would end up being:
build new debian-hyperkube-base (with CNI plugins v0.8.5) --> promote --> CNI v0.8.6 bump (which builds another new image) --> promote --> PR to use new debian-hyperkube-base image

liggitt commented on build/dependencies.yaml:
```diff
@@ -99,6 +101,20 @@ dependencies:
+      match: TAG\?=
+
+  - name: "k8s.gcr.io/debian-hyperkube-base: dependents"
+    version: 0.12.1
```
does this not match the `TAG=0.12.2` in the Makefile because we never released 0.12.2?

justaugustus replied:
Yep, that image appear to have never been built or pushed.

liggitt commented:
/lgtm
/approve

justaugustus commented:
Image build in progress here: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-kubernetes-push-image-debian-hyperkube-base/1265464966937317376 / https://console.cloud.google.com/cloud-build/builds/8fefaee2-3f03-4327-b641-3e134cb7f0d4?project=k8s-staging-build-image

justaugustus commented:
Image promotion PR: https://github.com/kubernetes/k8s.io/pull/899
